tests/notificationprofile/test_views.py

```diff
@@ -40,113 +38,236 @@ def setUp(self):
-        self.assertEqual(self.user1_rest_client.get(email_destination_path).status_code, status.HTTP_200_OK)
-        response = self.user1_rest_client.delete(email_destination_path)
         self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
-
-        self.assertEqual(self.user1_rest_client.get(email_destination_path).status_code, status.HTTP_404_NOT_FOUND)
+        self.assertFalse(DestinationConfig.objects.filter(media_id="email").filter(settings__synced=False).exists())
 
     def test_cannot_delete_synced_email_destination(self):
-        self.email_destination = self.user1.destinations.filter(media_id="email").filter(settings__synced=True).first()
-
-        email_destination_path = reverse(
-            "v2:notification-profile:destinationconfig-detail", args=[self.email_destination.pk]
+        self.assertTrue(DestinationConfig.objects.filter(media_id="email").filter(settings__synced=True).exists())
+        response = self.user1_rest_client.delete(
+            path=f"/api/v2/notificationprofiles/destinations/{self.synced_email_destination.pk}/"
         )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertTrue(DestinationConfig.objects.filter(media_id="email").filter(settings__synced=True).exists())
 
-        self.assertEqual(self.user1_rest_client.get(email_destination_path).status_code, status.HTTP_200_OK)
-        response = self.user1_rest_client.delete(email_destination_path)
+    def test_cannot_delete_connected_email_destination(self):
+        self.notification_profile1.destinations.add(self.non_synced_email_destination)
+
+        self.assertTrue(DestinationConfig.objects.filter(media_id="email").filter(settings__synced=False).exists())
+        response = self.user1_rest_client.delete(
+            path=f"/api/v2/notificationprofiles/destinations/{self.non_synced_email_destination.pk}/"
+        )
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertTrue(DestinationConfig.objects.filter(media_id="email").filter(settings__synced=False).exists())
 
-        self.assertEqual(self.user1_rest_client.get(email_destination_path).status_code, status.HTTP_200_OK)
+    def test_can_get_incidents_of_notification_profile(self):
+        response = self.user1_rest_client.get(
+            path=f"/api/v2/notificationprofiles/{self.notification_profile1.pk}/incidents/"
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["pk"], self.incident1.pk)
 
-    def test_cannot_delete_connected_email_destination(self):
-        self.email_destination = self.create_email_destination()
-        self.notification_profile1.destinations.add(self.email_destination)
+    def test_notification_profile_can_properly_change_timeslot(self):
```

Comment:
it is not clear what "properly" means, i changed this in the v1 version to be 
```suggestion
    def test_can_update_timeslot_for_notification_profile_with_valid_values(self):
```